src/engine_core/rendering/src/Vulkan/VulkanLoader.cpp

```diff
@@ -0,0 +1,111 @@
+#include "VulkanLoader.hpp"
+#include <fastgltf/parser.hpp>
+#include <fastgltf/types.hpp>
+#include <fastgltf/tools.hpp>
+#include "VkTypes.hpp"
+#include "VulkanRenderer.hpp"
+
+std::optional<std::vector<std::shared_ptr<Hush::MeshAsset>>> Hush::VulkanLoader::LoadGltfMeshes(VulkanRenderer* engine, std::filesystem::path filePath)
+{
+	fastgltf::GltfDataBuffer data;
+	data.loadFromFile(filePath);
+
+	constexpr fastgltf::Options loadingOptions = fastgltf::Options::LoadGLBBuffers | fastgltf::Options::LoadExternalBuffers;
+
+	fastgltf::Parser parser{};
+
+	fastgltf::Expected<fastgltf::Asset> loadedAsset = parser.loadBinaryGLTF(&data, filePath.parent_path(), loadingOptions);
+
+	HUSH_ASSERT(loadedAsset, "GLTF asset at {} not properly loaded, error: {}!", filePath.string(), fastgltf::getErrorMessage(loadedAsset.error()));
```

Comment:
Is it a fatal error or is it better to return a Result?

Reply:
This should return a result, you're right, I'll be implementing that